code/modules/mob/living/simple_animal/friendly/trader.dm

```diff
@@ -0,0 +1,242 @@
+	if(!cash || cash.credits < products[item_to_buy])
+		say(nocashphrase)
+		return
+	cash.spend(products[item_to_buy])
+	var/obj/item/bought_item = new item_to_buy(loc)
+	user.put_in_hands(bought_item)
+	playsound(src, sell_sound, 50, TRUE)
+	say(buyphrase)
+
+/**
+ * Checks if an item is in the list of wanted items and if it is after a Yes/No radial returns generate_cash with the value of the item for the NPC
+ * Arguments:
+ * * user - The mob trying to sell something
+ * * selling - The item being sold
+ */
+/mob/living/simple_animal/hostile/retaliate/trader/proc/sell_item(mob/user, selling)
+	var/obj/item/sellitem = selling
+	var/progressive_type = ""
+	var/cost
+	if(!sellitem)
+		return FALSE
+	for(var/type_level in splittext("[sellitem.type]","/"))
+		if(type_level == "")
+			continue
+		progressive_type += ("/"+type_level)
+		if(text2path(progressive_type) in wanted_items)
+			cost = wanted_items[text2path(progressive_type)]
+	if(!cost)
+		return FALSE
+	say(interestedphrase)
+	to_chat(user, "<span class='notice'>You will receive [cost] credits for each one of [sellitem].</span>")
+	var/list/npc_options = list(
+		"Yes" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_yes"),
+		"No" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_no")
+		)
+	var/npc_result = show_radial_menu(user, src, npc_options, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
+	face_atom(user)
+	if(npc_result != "Yes")
+		say(itemsellcancelphrase)
+		return TRUE
+	say(itemsellacceptphrase)
+	playsound(src, sell_sound, 50, TRUE)
+	if(isstack(sellitem))
+		var/obj/item/stack/stackoverflow = sellitem
+		log_econ("[stackoverflow] has been sold to [src] by [user] for [cost * stackoverflow.amount] cash.")
+		generate_cash(cost * stackoverflow.amount, user)
+		stackoverflow.use(stackoverflow.amount)
+		return TRUE
+	log_econ("[sellitem] has been sold to [src] by [user] for [cost] cash.")
+	generate_cash(cost, user)
+	qdel(sellitem)
+	return TRUE
+
+/**
+ * Creates a holochip the value set by the proc and puts it in the user's hands
+ * Arguments:
+ * * value - The amount of cash that will be on the holochip
+ * * user - The mob we put the holochip in hands of
+ */
+/mob/living/simple_animal/hostile/retaliate/trader/proc/generate_cash(value, mob/user)
+	var/obj/item/holochip/chip = new /obj/item/holochip(get_turf(user), value)
+	user.put_in_hands(chip)
+
+/mob/living/simple_animal/hostile/retaliate/trader/mrbones
+	name = "Mr. Bones"
+	desc = "A skeleton merchant, he seems very humerus."
+	speak_emote = list("rattles")
+	speech_span = SPAN_SANS
+	sell_sound = 'sound/voice/hiss2.ogg'
+	mob_biotypes = MOB_UNDEAD|MOB_HUMANOID
+	products = list(/obj/item/clothing/head/helmet/skull = 150,
+					/obj/item/clothing/mask/bandana/skull = 50,
+					/obj/item/food/cookie/sugar/spookyskull = 10,
+					/obj/item/instrument/trombone/spectral = 10000,
+					/obj/item/shovel/serrated = 150
+					)
```

Comment:
Fix the indenting the same way I showed before. Only indent up one level, and dedent the parenthese.